Implement process state filter (#1544/#1755)

BenBE opened the pull request:
This PR is a continuation of #1755, due to an accidentally fumbled push operation.

@melrovieira: Sorry, the git push to edit your PR (to clean up the commits), accidentally pushed the wrong commit to Github and thus unintentionally closed the PR.

This PR restores the actually intended push.

melrovieira commented:
> Este PR é uma continuação do #1755 , devido a uma operação de push malsucedida acidentalmente.
> 
> @melrovieira:Desculpe, o git push para editar seu PR (para limpar os commits) acidentalmente enviou o commit errado para o Github e, assim, fechou o PR sem querer.
> 
> Este PR restaura o push realmente pretendido.

No problem, do you want me to fix the indentation?

BenBE commented:
@melrovieira: Não precisa; já corrigi o recuo e limpei tudo. Por favor, dê uma olhada nos commits atualizados neste PR. Tomei a liberdade de limpá-los para conter apenas alterações em locais do código onde a alteração era realmente necessária. Em PRs futuros, certifique-se de usar `git add -p` para limitar as modificações a locais no código que realmente pertencem às alterações necessárias para o recurso. O guia de estilo é um pouco flexível sobre isso, mas a intenção é, ao executar o comando astyle, selecionar as alterações realmente relevantes. Isso é um pouco complicado pelo fato de que a base de código original misturou o estilo de indentação para instruções switch/case em relação à indentação.

---
@melrovieira: No need; I already fixed the indentation and cleaned things up. Please take a look at the updated commits in this PR. I took the liberty to clean them up to only contain changes to code locations where a change was actually necessary. In future PRs be sure to use `git add -p` to limit modifications to locations in the code that actually belong to the changes necessary for the feature. The style guide is somewhat loose on this, but the intention is to, when running the astyle command, pick the actually relevant changes. This is somewhat complicated by the fact, that the original codebase mixed the indentation style for switch/case statements regarding indentation.

melrovieira commented:
> @melrovieira: Não precisa; já corrigi o recuo e limpei tudo. Por favor, dê uma olhada nos commits atualizados neste PR. Tomei a liberdade de limpá-los para conter apenas alterações em locais do código onde a alteração era realmente necessária. Em PRs futuros, certifique-se de usar `git add -p` para limitar as modificações a locais no código que realmente pertencem às alterações necessárias para o recurso. O guia de estilo é um pouco flexível sobre isso, mas a intenção é, ao executar o comando astyle, selecionar as alterações realmente relevantes. Isso é um pouco complicado pelo fato de que a base de código original misturou o estilo de indentação para instruções switch/case em relação à indentação.
> 
> @melrovieira: No need; I already fixed the indentation and cleaned things up. Please take a look at the updated commits in this PR. I took the liberty to clean them up to only contain changes to code locations where a change was actually necessary. In future PRs be sure to use `git add -p` to limit modifications to locations in the code that actually belong to the changes necessary for the feature. The style guide is somewhat loose on this, but the intention is to, when running the astyle command, pick the actually relevant changes. This is somewhat complicated by the fact, that the original codebase mixed the indentation style for switch/case statements regarding indentation.

Okay, I'll remember that. Thank you very much. 🤝

natoscott commented on CommandLine.c:
```diff
@@ -255,6 +259,33 @@ static CommandLineStatus parseArguments(int argc, char** argv, CommandLineSettin
+            for (char* c = optarg; *c != '\0'; c++) {
+               bool valid = false;
+
+               for (ProcessState s = UNKNOWN; s <= SLEEPING; s++) {
```
Allow a comma/space-separated list here?  Might make for a better user experience on the command line.

Just a "if (*c == ',' || isspace(*c)) continue;" in the outer loop should do it.

natoscott commented on Process.h:
```diff
@@ -335,6 +335,8 @@ void Process_makeCommandStr(Process* this, const struct Settings_ *settings);
 void Process_writeCommand(const Process* this, int attr, int baseAttr, RichString* str);
 
+char processStateChar(ProcessState state);
+
```
Since this is being made available globally now, I think a name matching the rest of the header declarations would be ideal (e.g. Process_stateChar).
